core/server/master/src/main/java/alluxio/master/file/contexts/CreateDirectoryContext.java

```diff
@@ -108,6 +109,14 @@ public CreateDirectoryContext setDefaultAcl(List<AclEntry> defaultAcl) {
     return mDefaultAcl;
   }
 
+  @Override
+  public OperationId getOperationId() {
+    if (getOptions().hasCommonOptions() && getOptions().getCommonOptions().hasOperationId()) {
+      return OperationId.fromFsProto(getOptions().getCommonOptions().getOperationId());
+    }
+    return super.getOperationId();
```

Comment:
Will this return a `null` from `super.getOperationId()` and possibly introduce an NPE for someone else?

Reply:
Right. All code path is marked with @Nullable annotation. I have now added where it was missing as well.